### PR TITLE
Fix System/Boolean.xml code snippet

### LINKED ISSUE #4368: Missing & incorrect examples in xml/System/Boolean.xml
### Missing example

The following paragraph says that an example follows, but this example is absent:

> When converting from floating-point values to Boolean values, the conversion methods perform an exact comparison with zero. If the floating-point value has lost precision, the result can be unexpected. This is illustrated in the following example, in which a Double variable whose value should be zero is converted to a Boolean value. As the example shows, the result is true because repeated additions of 0.2 have resulted in a loss of precision.

### Incorrect example

The wrong example is in the page _following_ this excerpt:

> However, when you want to compare a Boolean variable with the literal Boolean value true or false, it is not necessary to do an explicit comparison, because the result of evaluating a Boolean value is that Boolean value. For example, the expressions
>
> `if (booleanValue) {`
>
> and

Presumably the example should be

    if (booleanValue == true) {

However it is:

```
using System;

public class Example
{
   public static void Main()
   {
      bool[] hasServiceCharges = { true, false };
      Decimal subtotal = 120.62m;
      Decimal shippingCharge = 2.50m;
      Decimal serviceCharge = 5.00m;

      foreach (var hasServiceCharge in hasServiceCharges) {
         Decimal total = subtotal + shippingCharge +
                                (hasServiceCharge ? serviceCharge : 0);
         Console.WriteLine("hasServiceCharge = {1}: The total is {0:C2}.",
                           total, hasServiceCharge);
      }
   }
}
// The example displays output like the following:
//       hasServiceCharge = True: The total is $128.12.
//       hasServiceCharge = False: The total is $123.12.
```

### PR DESCRIPTION
**Pull request by glihm**
## Summary

Fix System.Boolean snippet references and typo in csharp code. VB code is fine.

[Edited by gewarren to change "fixes" to "contributes", since there is still a missing example to fix in that issue.]
Contributes to #4368

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/dotnet-api-docs?pullRequest=4551) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/dotnet-api-docs?pullRequest=4551) before we can accept your contribution.<br/><br/>:x: glihm [sign now](https://cla.dotnetfoundation.org/dotnet/dotnet-api-docs?pullRequest=4551)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/dotnet-api-docs?pullRequest=4551) it.</sub>

File: xml/System/Boolean.xml [hunk 1]
```diff
@@ -161,13 +161,13 @@
...
- [!code-csharp[System.Boolean.Structure#12](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.boolean.structure/cs/operations1.cs#12)]
- [!code-vb[System.Boolean.Structure#12](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.boolean.structure/vb/operations1.vb#12)]
+ [!code-csharp[System.Boolean.Structure#11](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.boolean.structure/cs/operations1.cs#11)]
+ [!code-vb[System.Boolean.Structure#11](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.boolean.structure/vb/operations1.vb#11)]
 
  and
 
- [!code-csharp[System.Boolean.Structure#13](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.boolean.structure/cs/operations2.cs#13)]
- [!code-vb[System.Boolean.Structure#13](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.boolean.structure/vb/operations2.vb#13)]
+ [!code-csharp[System.Boolean.Structure#12](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.boolean.structure/cs/operations2.cs#12)]
```
Comment (gewarren) [inline xml/System/Boolean.xml:169]:
```suggestion
 [!code-csharp[System.Boolean.Structure#12](~/samples/snippets/csharp/VS_Snippets_CLR_System/system.boolean.structure/cs/operations1.cs#12)]
```

File: xml/System/Boolean.xml [hunk 2]
```diff
@@ -161,13 +161,13 @@
...
+ [!code-vb[System.Boolean.Structure#12](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.boolean.structure/vb/operations2.vb#12)]
```
Comment (gewarren) [inline xml/System/Boolean.xml:170]:
```suggestion
 [!code-vb[System.Boolean.Structure#12](~/samples/snippets/visualbasic/VS_Snippets_CLR_System/system.boolean.structure/vb/operations1.vb#12)]
```

**Comment by glihm** [comment]
> @glihm This looks great except for a small tweak in the file name. Preview link for internal folks: https://review.docs.microsoft.com/en-us/dotnet/api/system.boolean?branch=pr-en-us-4551&view=netcore-3.1#comparing-boolean-values.

@gewarren nice catch! Thank you!
(I am not internal I can't see the link) :)

**Pull request merged**